build_wheels.py

```diff
@@ -172,28 +177,239 @@ def assemble_requirements(idf_branches: List[str], idf_constraints: List[str]) -
+        - print_requirements = true will print the changes
+    """
+    new_assembled_requirements = set()
+    not_in_exclude = []
+    if print_requirements:
+        print(Fore.BLUE + '---------- REQUIREMENTS ----------')
+
+    for requirement in assembled_requirements:
+        printed = False
+        for req in exclude_list:
+            if req.name not in requirement.name:
+                not_in_exclude.append(True)
+
+            if req.name in requirement.name:
+
+                if not req.specifier and not req.marker:
+                    if print_requirements:
+                        print(Fore.RED + f'-- {requirement}')
+                    continue
+
+                if requirement.specifier and req.specifier:
+                    new_specifier = requirement.specifier & req.specifier
+                elif requirement.specifier and not req.specifier:
+                    new_specifier = requirement.specifier
+                elif not requirement.specifier and req.specifier:
+                    new_specifier = req.specifier
+                else:
+                    new_specifier = ''
+
+                if requirement.marker and req.marker:
+                    new_markers = f'({requirement.marker}) and ({req.marker})'
+                elif requirement.marker and not req.marker:
+                    new_markers = requirement.marker
+                elif not requirement.marker and req.marker:
+                    new_markers = req.marker
+                else:
+                    new_markers = ''
+
+
+                if new_markers:
+                    new_requirement = Requirement(f'{requirement.name}{new_specifier}; {new_markers}')
+                else:
+                    new_requirement = Requirement(f'{requirement.name}{new_specifier}')
+
+                new_assembled_requirements.add(new_requirement)
+
+                if print_requirements:
+                    if not printed:
+                        print(Fore.RED + f'-- {requirement}')
+                        printed = True
+                    print(Fore.GREEN + f'++ {new_requirement}')
+
+
+        if len(not_in_exclude) == len(exclude_list):
+            if print_requirements:
+                print(Style.RESET_ALL + str(requirement))
+            new_assembled_requirements.add(requirement)
+
+        not_in_exclude.clear()
+
+    if print_requirements:
+        print(Fore.BLUE + '---------- END OF REQUIREMENTS ----------' + Style.RESET_ALL)
+
+    return new_assembled_requirements
+
+
+# --- Build wheels ---
+failed_wheels = 0
+succeeded_wheels = 0
+
+def build_wheels(requirements: set):
+    """Build Python wheels"""
+    global failed_wheels, succeeded_wheels
+
+    dir = f'{os.path.curdir}{(os.sep)}downloaded_wheels'
+    for requirement in requirements:
+        # --only-binary requirement wheel build
+        if non_classic_requirement and non_classic_requirement[0].replace('--only-binary ', '') in requirement.name:
```

Comment:
Na name non_classic suggest something generic. You've mentioned only-binary as an example. But here `--only-binary` is hardcoded three times. Perhaps we need to make this general (compatible with other similar cases), even only now only-binary will be defined there.